lib/charms/layer/apache_zeppelin.py

```diff
@@ -168,3 +170,22 @@ def close_ports(self):
     def cleanup(self):
         self.dist_config.remove_dirs()
         unitdata.kv().set('zeppelin.installed', False)
+
+
+class ZeppelinAPI(object):
+    """
+    Helper for interacting with the Appache Zeppelin REST API.
+    """
+    def _url(self, *parts):
+        url = 'http://localhost:9090/api/'
```

Comment:
Since these requests are coming in over the relation, they will apply to all units, which is the right thing to do.  (Every unit needs a copy of the notebook or interpreter settings.)